layers/+lang/ruby/packages.el

```diff
@@ -128,6 +128,7 @@
         (spacemacs/declare-prefix-for-mode mode "mg" "ruby/goto")
         (spacemacs/declare-prefix-for-mode mode "mh" "ruby/docs")
         (spacemacs/declare-prefix-for-mode mode "ms" "ruby/repl")
+        (add-hook 'compilation-filter-hook 'inf-ruby-auto-enter) ;; automatically enters inf-ruby mode from a compilation buffer
```

Comment:
You can make the hook local to the current buffer with the last argument of `add-hook`.
So what can be done to limit it to ruby buffer is to wrap the `add-hook` into a function and add this function to `ruby-mode-hook`.